ql/indexes/inflationindex.cpp

```diff
@@ -70,23 +70,46 @@ namespace QuantLib {
                       Frequency frequency,
                       const Period& availabilityLag,
                       const Currency& currency,
-                      const Handle<ZeroInflationTermStructure>& zeroInflation)
-    : InflationIndex(familyName, region, revised, interpolated,
+                      const Handle<ZeroInflationTermStructure>& zeroInflation,
+                      bool forecastWhenPossible)
+        : InflationIndex(familyName, region, revised, interpolated,
                      frequency, availabilityLag, currency),
-      zeroInflation_(zeroInflation) {
+      zeroInflation_(zeroInflation),
+      forecastWhenPossible_(forecastWhenPossible) {
         registerWith(zeroInflation_);
     }
 
+    bool ZeroInflationIndex::forecastTodaysFixing(const Date& fixingDate) const {
+        return forecastWhenPossible_ && fixingDate >= zeroInflationTermStructure()->baseDate();
+    }
+
+
+    Rate ZeroInflationIndex::fixing(const Date& fixingDate) const {
+        return fixing(fixingDate, fixingDate);
+    }
+
     Rate ZeroInflationIndex::fixing(const Date& aFixingDate,
-                                    bool /*forecastTodaysFixing*/) const {
-        if (!needsForecast(aFixingDate)) {
+                                    bool forecastTodaysFixing) const {
+        return fixing(aFixingDate, aFixingDate, forecastTodaysFixing);
+    }
+
+    Rate ZeroInflationIndex::fixing(const Date& fixingDate,
+                                    const Date& referenceDate) const {
+        return fixing(fixingDate, referenceDate, forecastTodaysFixing(fixingDate));
+    }
+    
+    Rate ZeroInflationIndex::fixing(const Date& aFixingDate,
+                                    const Date& referenceDate,
+                                    bool forecastTodaysFixing) const {
+        if (!forecastTodaysFixing && !needsForecast(aFixingDate)) {
             std::pair<Date,Date> lim = inflationPeriod(aFixingDate, frequency_);
             const TimeSeries<Real>& ts = timeSeries();
             Real pastFixing = ts[lim.first];
             QL_REQUIRE(pastFixing != Null<Real>(),
                        "Missing " << name() << " fixing for " << lim.first);
             Real theFixing = pastFixing;
-            if (interpolated_) {
+            //Interpolate only when necessary
+            if (interpolated_ && referenceDate > lim.first) {
                 // fixings stored on first day of every period
                 if (aFixingDate == lim.first) {
```

Comment:
Is this still true? If `aFixingDate` equals `lim.first` but the reference date is in the middle of `reference_period_lim`, should we interpolate or not?